pkl-core/src/test/kotlin/org/pkl/core/project/ProjectTest.kt

```diff
@@ -116,7 +117,10 @@ class ProjectTest {
     """.trimIndent())
     val project = Project.loadFromPath(projectPath)
     assertThat(project.`package`).isEqualTo(expectedPackage)
-    assertThat(project.settings).isEqualTo(expectedSettings)
+    assertThat(project.settings)
+      .usingRecursiveComparison()
+      .withComparatorForType(compareBy(Pattern::toString), Pattern::class.java)
+      .isEqualTo(expectedSettings)
```

Comment:
We should implement a custom `equals` and `hashCode` method in `PklEvaluatorSettings`; we _should_ be able to use `equals` to compare these two.